spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/MonitoringDashboardInfoProperties.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.config;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.dataflow.core.DataFlowPropertyKeys;
+import org.springframework.cloud.dataflow.rest.resource.about.MonitoringDashboardType;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author Christian Tzolov
+ */
+@ConfigurationProperties(prefix = MonitoringDashboardInfoProperties.VERSION_INFO_PREFIX)
+public class MonitoringDashboardInfoProperties {
+
+	public static final String VERSION_INFO_PREFIX = DataFlowPropertyKeys.PREFIX + "monitoring-info";
+
+	/**
+	 * Root URL to access the monitoring dashboards
+	 */
+	private String url = "";
+
+	/**
+	 * If provided, can be used to authenticate with Grafana.
+	 * https://docs.grafana.org/http_api/auth/#create-api-token
+	 *
+	 * The 'Authorization' header value should be: 'Bearer (your api key)'.
+	 */
+	private String token = "";
+
+	/**
+	 * Dashboard refresh interval in Seconds
+	 */
+	private int refreshInterval = 15;
+
+	/**
+	 *
+	 */
+	private MonitoringDashboardType dashboardType = MonitoringDashboardType.grafana;
+
+	/**
+	 *
+	 */
+	@Value("${management.metrics.export.wavefront.source:default-scdf-source}")
```

Comment:
I understand that currently we only have the `source` concept for wavefront, but having wavefront specific key here may be avoided by moving wavefront specific fields into a subtype?

Reply:
At the moment the source is the only diverging property. And it is hard to justify a separate class.  Furthermore for the 2.6.x we have the deprecated Grafana related classes. Once we move to 2.7 and remove the deprecated classes we can rethink should we have pre-dashboarding class. 
Alternative is to generalized even further and use the `Map<String, String> specific` property that would allow defining dynamically the specific properties. I actually like this approach better. But we should discuss it when moving to 2.7.x line.